kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/general/SetTypeRefCommand.java

```
@@ -0,0 +1,98 @@
+import org.kie.workbench.common.stunner.core.client.command.CanvasCommandResultBuilder;
+import org.kie.workbench.common.stunner.core.client.command.CanvasViolation;
+import org.kie.workbench.common.stunner.core.command.Command;
+import org.kie.workbench.common.stunner.core.command.CommandResult;
+import org.kie.workbench.common.stunner.core.graph.command.GraphCommandExecutionContext;
+import org.kie.workbench.common.stunner.core.graph.command.GraphCommandResultBuilder;
+import org.kie.workbench.common.stunner.core.graph.command.impl.AbstractGraphCommand;
+import org.kie.workbench.common.stunner.core.rule.RuleViolation;
+
+public class SetTypeRefCommand extends AbstractCanvasGraphCommand implements VetoExecutionCommand,
+                                                                             VetoUndoCommand {
+
+    private final HasVariable hasVariable;
+    private final QName typeRef;
+    private final org.uberfire.mvp.Command canvasOperation;
+
+    private final Optional<QName> oldTypeRef;
+
+    public SetTypeRefCommand(final HasVariable hasVariable,
+                             final QName typeRef,
+                             final org.uberfire.mvp.Command canvasOperation) {
+        this.hasVariable = hasVariable;
+        this.typeRef = typeRef;
+        this.canvasOperation = canvasOperation;
+
+        this.oldTypeRef = Optional.ofNullable(hasVariable.getVariable().getTypeRef());
+    }
+
+    @Override
+    protected Command<GraphCommandExecutionContext, RuleViolation> newGraphCommand(final AbstractCanvasHandler handler) {
+        return new AbstractGraphCommand() {
+            @Override
+            protected CommandResult<RuleViolation> check(final GraphCommandExecutionContext gce) {
+                return GraphCommandResultBuilder.SUCCESS;
+            }
+
+            @Override
+            public CommandResult<RuleViolation> execute(final GraphCommandExecutionContext gce) {
+                hasVariable.getVariable().setTypeRef(typeRef);
```

is `hasVariable.getVariable()` always non-null?

---

Hi @jomarko it is guaranteed with the existing code; but better not to assume it'd be like that in new code so I've added a commit to fail fast in the event of ```null``` parameters.